gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -717,3 +729,284 @@ message TranslatedString {
+  enum WheelchairAccessibleStatus {
+    UNKNOWN_WHEELCHAIR_ACCESSIBILITY = 0;
+    WHEELCHAIR_ACCESSIBLE = 1;
+    NOT_WHEELCHAIR_ACCESSIBLE = 2;
+  }
+
+  // Options for biked_allowed:
+  // 0 - No bike information for the trip.
+  // 1 - Vehicle being used on this particular trip can accommodate at least one bicycle.
+  // 2 - No bicycles are allowed on this trip.
+  enum BikesAllowedStatus {
+    UNKNOWN_BIKES_ALLOWANCE = 0;
+    BIKES_ALLOWED = 1;
+    BIKES_NOT_ALLOWED = 2;
+  }
+}
+
+// Provides the updated values for the stop time.
+message StopTimeProperties {
+  // Refers to a stop_id defined in the GTFS.
+  optional string platform_id = 1;
+  // Text that appears on signage identifying the trip's destination to riders.
+  optional TranslatedString stop_headsign = 2;
+  // Indicates pickup method.
+  optional PickupType pickup_type = 3 [default = REGULAR_PICKUP];
+  // Indicates drop off method.
+  optional DropOffType drop_off_type = 4 [default = REGULAR_DROP_OFF];
+  // Actual distance traveled along the associated shape, from the first stop to the stop specified in this record.
+  optional float shape_dist_traveled = 5;
+
+  // Valid options for pickup_type:
+  enum PickupType {
+    REGULAR_PICKUP = 0;
+    NO_PICKUP = 1;
+    MUST_PHONE_AGENCY_PICKUP = 2;
+    MUST_ASK_DRIVER_PICKUP = 3;
+  }
+
+  // Valid options for drop_off_type:
+  enum DropOffType {
+    REGULAR_DROP_OFF = 0;
+    NO_DROP_OFF = 1;
+    MUST_PHONE_AGENCY_DROP_OFF = 2;
+    MUST_ASK_DRIVER_DROP_OFF = 3;
+  }
+}
+
+// Describes new trips.
+message Trip {
+  // Identifier of the trip. Must be different than any trip_id defined in the GTFS.
+  required string trip_id = 1;
+  // Identifies a route.
+  optional string route_id = 2;
+  // Text that appears on signage identifying the trip's destination to riders.
+  required TranslatedString trip_headsign = 3;
+  // Public facing text used to identify the trip to riders, for instance, to identify train numbers for commuter rail trip
+  required string trip_short_name = 4;
+  // Indicates the direction of travel for a trip.
+  optional uint32 direction_id = 5;
+  // Identifies the block to which the trip belongs.
+  optional string block_id = 6;
+  // Identifies a geospatial shape describing the vehicle travel path for a trip.
+  optional string shape_id = 7;
+  // Indicates wheelchair accessibility.
+  optional WheelchairAccessibleStatus wheelchair_accessible = 8 [default = UNKNOWN_WHEELCHAIR_ACCESSIBILITY];
+  // Indicates whether bikes are allowed or not.
+  optional BikesAllowedStatus biked_allowed = 9 [default = UNKNOWN_BIKES_ALLOWANCE];
+  //  Defines which trips are being replaced by this trip.
+  repeated string replaces_trip_id = 10;
+  // The arrival and departure times for this trip. Entities must be ordered by stop_sequence.
+  repeated StopTime stop_time = 11;
+  // Date on which the trip will be run. Must be provided in YYYYMMDD format.
+  required string start_date = 12;
```

Comment:
What about making this `repeated`? If the same trip is running on two days, it seems like a waste to definite everything (including the stop times) twice.

Reply:
Yeah, I see what you mean. Benefit would be to potentially halve the size of the PB file if you have the same detour(s) running two days back-to-back.

However, it would break our existing design goal of having a 1:1 `Trip`-to-scheduled trip representation, though. This is partly to discourage producers from throwing a ton of trips into the future in the protobuf feed that should really be in static GTFS CSVs. Thinking quickly about how frequency-based (exact_times=0) trips would work in ServiceChanges, I think the same issues apply.

How many dates into the future would you see MBTA using this? In other words, what would be the typical size of this array if it were `repeated`?

@timMillet @LeoFrachet Thoughts on this?

Reply:
My notes from earlier discussions peg data in ServiceChanges as representing changes for the next 3-5 days.

Reply:
Per further discussion with @timMillet and @LeoFrachet today, we're going to make `start_date` `repeated` for new Trips 👍 . I just made the change in https://github.com/MobilityData/transit/pull/47#discussion_r406298976.